lib/msal-node/src/network/HttpClient.ts

```diff
@@ -23,21 +25,39 @@ export class HttpClient implements INetworkModule {
      */
     async sendGetRequestAsync<T>(
         url: string,
-        options?: NetworkRequestOptions
+        options?: NetworkRequestOptions,
+        cancellationToken?: number,
+        proxy?: string
     ): Promise<NetworkResponse<T>> {
-        const request: AxiosRequestConfig = {
-            method: HttpMethod.GET,
-            url: url,
-            headers: options && options.headers,
-            validateStatus: () => true
-        };
+        let response;
+        let responseObject: any = {};
+
+        if (proxy) {
+            const fetchOptions =  {
+                headers: options?.headers,
+                body: options?.body,
+                agent: new HttpsProxyAgent(proxy),
+            };
+    
+            response = await fetch(url, fetchOptions);
```

Comment:
The other option is to document how applications can provide their own http client, which give them complete control over the proxied requests. Did we consider that?

Reply:
Yes. I am about to suggest that. The alternative I have in mind is provide a sample with a network client that supports proxies and leave it at that with instructions. We are open to suggestions.

Reply:
Per our discussion, we are switching from axios to node-fetch.